library/src/main/java/com/google/android/exoplayer/util/BitArray.java

```diff
@@ -151,7 +151,7 @@ public int readUnsignedByte() {
     byte b;
```

Comment:
Your code's return value is not correct, it actually needs to `return value & 0xFF`